python/lsst/ts/Dome/dome_csc.py

```diff
@@ -188,21 +208,26 @@ async def write_then_read_reply(self, **params):
          """
         cmd = params["command"]
         st = encoding_tools.encode(**params)
-        self.log.info(f"Sending command {st}")
-        self.writer.write(st.encode() + b"\r\n")
-        await self.writer.drain()
-        read_bytes = await asyncio.wait_for(self.reader.readuntil(b"\r\n"), timeout=1)
-        data = encoding_tools.decode(read_bytes.decode())
-
-        response = data["response"]
-        if response > ResponseCode.OK:
-            self.log.error(f"Received ERROR {data}.")
-            if response == ResponseCode.INCORRECT_PARAMETER:
-                raise ValueError(f"The command {cmd} contains an incorrect parameter.")
-            elif response == ResponseCode.UNSUPPORTED_COMMAND:
-                raise KeyError(f"The command {cmd} is unsupported.")
-
-        return data
+        async with self.communication_lock:
+            self.log.debug(f"Sending command {st}")
+            self.writer.write(st.encode() + b"\r\n")
+            await self.writer.drain()
+            read_bytes = await asyncio.wait_for(
+                self.reader.readuntil(b"\r\n"), timeout=1
```

Comment:
Please consider a named constants for the timeout, so it's visible near the top of the file (with the other constants).

Reply:
There already was one so I am using that one now.